app/models/concerns/scripts/import.rb

```diff
@@ -5,14 +5,24 @@ module Scripts::Import
 
   module ClassMethods
     def import zip_path
-      transaction { import_zip zip_path }
+      transaction do
+        exceptions = {}
+        import_zip zip_path, exceptions
+        raise ActiveRecord::ActiveRecordError, (list_of_exceptions exceptions) if exceptions.present?
+      end
```

Comment:
Esto lo podés hacer distinto, para no tener que meter en el error los mensajes. Algo así:
```suggestion
      exceptions = {}

      transaction do
        exceptions = import_zip zip_path

        raise ActiveRecord::Rollback if exceptions.present?
      end

      exceptions
```

De esta forma se deshace la transacción, y por el tipo de excepción es lo único que hace (no explota al método). Entonces podés devolver tranquilo el hash con los errores.

Lo dejé con `exceptions` para que se entienda el cambio, pero le cambiaría el nombre, me suena un toque mejor `errors`.

Reply:
Bien. Tomo la idea y devuelvo siempre el arreglo de scripts con problemas y lo analizo en el controller para ver que vista utilizamos.